packages/core/src/lib/history.unit.test.ts

```diff
@@ -0,0 +1,160 @@
+
+    await history(historyOptions, ['abc']);
+    expect(collectAndPersistReports).toHaveBeenCalledTimes(1);
+    expect(collectAndPersistReports).toHaveBeenNthCalledWith(
+      1,
+      expect.objectContaining({
+        persist: expect.objectContaining({
+          filename: 'abc-report',
+          format: ['json'],
+        }),
+      }),
+    );
+  });
+
+  it('should call upload by default', async () => {
+    const historyOptions: HistoryOptions = {
+      ...(MINIMAL_CONFIG_MOCK as HistoryOptions),
+      upload: {
+        server: 'https://server.com/api',
+        project: 'cli',
+        apiKey: '1234',
+        organization: 'code-pushup',
+        timeout: 4000,
+      },
+    };
+    await history(historyOptions, ['abc']);
+
+    expect(upload).toHaveBeenCalledTimes(1);
+    expect(upload).toHaveBeenNthCalledWith(
+      1,
+      expect.objectContaining({
+        persist: expect.objectContaining({ filename: 'abc-report' }),
+      }),
+    );
+  });
+
+  it('should not call upload if skipUploads is set to false', async () => {
+    const historyOptions: HistoryOptions = {
+      ...(MINIMAL_CONFIG_MOCK as HistoryOptions),
+      upload: {
+        server: 'https://server.com/api',
+        project: 'cli',
+        apiKey: '1234',
+        organization: 'code-pushup',
+        timeout: 4000,
+      },
+      skipUploads: true,
+    };
+    await history(historyOptions, ['abc']);
+
+    expect(upload).not.toHaveBeenCalled();
+  });
+
+  it('should not call upload if upload config is not given', async () => {
+    const historyOptions: HistoryOptions = {
+      ...(MINIMAL_CONFIG_MOCK as HistoryOptions),
+    };
+    await history(historyOptions, ['abc']);
+
+    expect(upload).not.toHaveBeenCalled();
+  });
+});
+
+describe('prepareHashes', () => {
+  it('should return commit hashes in reverse order', () => {
+    expect(
+      prepareHashes({
+        all: [
+          {
+            hash: '22287eb716a84f82b5d59e7238ffcae7147f707a',
+            date: 'Thu Mar 7 20:13:33 2024 +0100',
+            message:
+              'test: replace default (buggy on Windows) with basic reporter',
+            refs: 'string',
+            body: 'string',
+            author_name: 'string',
+            author_email: 'string',
+          },
+          {
+            hash: '111b284e48ddf464a498dcf22426a9ce65e2c01c',
+            date: 'Thu Mar 7 20:13:34 2024 +0100',
+            message: 'chore: exclude fixtures from ESLint',
+            refs: 'string',
+            body: 'string',
+            author_name: 'string',
+            author_email: 'string',
+          },
+        ],
+        total: 2,
+        latest: {
+          hash: '22287eb716a84f82b5d59e7238ffcae7147f707a',
+          date: 'Thu Mar 7 20:13:33 2024 +0100',
+          message:
+            'test: replace default (buggy on Windows) with basic reporter',
+          refs: 'string',
+          body: 'string',
+          author_name: 'string',
+          author_email: 'string',
```

Comment:
The test data could use some love. The aim is to simulate realistic scenarios.

```suggestion
            hash: '22287eb716a84f82b5d59e7238ffcae7147f707a',
            date: 'Thu Mar 7 20:13:33 2024 +0100',
            message:
              'test: change test reported to basic in order to work on Windows',
            refs: 'string',
            body: '',
            author_name: 'John Doe',
            author_email: 'john.doe@gmail.com',
          },
          {
            hash: '111b284e48ddf464a498dcf22426a9ce65e2c01c',
            date: 'Thu Mar 7 20:13:34 2024 +0100',
            message: 'chore: exclude fixtures from ESLint',
            refs: 'string',
            body: '',
            author_name: 'Jane Doe',
            author_email: 'jane.doe@gmail.com',
          },
        ],
        total: 2,
        latest: {
          hash: '22287eb716a84f82b5d59e7238ffcae7147f707a',
          date: 'Thu Mar 7 20:13:33 2024 +0100',
          message:
            'test: change test reported to basic in order to work on Windows',
          refs: 'string',
          body: '',
          author_name: 'John Doe',
          author_email: 'john.doe@gmail.com',
```

Reply:
Thx for the love ❤️